src/cli/task/deps.rs

```
@@ -0,0 +1,155 @@
+use crate::{
+    config::{Config, Settings},
+    task::{Deps, Task},
+    ui::{style, tree::print_tree},
+};
+use console::style;
+use itertools::Itertools;
+use miette::Result;
+use petgraph::dot::Dot;
+
+/// [experimental] Display a tree visualization of a dependency graph
+#[derive(Debug, clap::Args)]
+#[clap(verbatim_doc_comment, after_long_help = AFTER_LONG_HELP)]
+pub struct TaskDeps {
+    /// Tasks to get dependencies for
+    /// Can specify multiple tasks by separating with spaces
+    /// e.g.: mise task deps task1 task2
+    #[clap(verbatim_doc_comment)]
+    pub tasks: Option<Vec<String>>,
+
+    /// Display dependencies in DOT format
+    #[clap(long, alias = "dot", verbatim_doc_comment)]
+    pub dot: bool,
+}
+
+impl TaskDeps {
+    pub fn run(self) -> Result<()> {
+        let config = Config::try_get()?;
+        let settings = Settings::try_get()?;
+        settings.ensure_experimental()?;
+
+        let tasks = if self.tasks.is_none() {
+            self.get_all_tasks(&config)?
+        } else {
+            self.get_task_lists(&config)?
+        };
+
+        if self.dot {
+            let _ = self.print_deps_dot(&config, tasks);
+        } else {
+            let _ = self.print_deps_tree(&config, tasks);
```

not clear to me why you're swallowing the error here, if you can't figure something out with rust lmk

---

good catch